featurebyte/api/feature_job_setting_analysis.py

```diff
@@ -142,7 +146,7 @@ def download_report(self, output_path: Optional[Union[str, Path]] = None) -> Pat
         output_path = output_path or Path(f"./{file_name}")
         output_path = Path(output_path)
 
-        if output_path.exists():
+        if output_path.exists() and not overwrite:
             raise FileExistsError(f"{output_path} already exists.")
 
         with open(output_path, "wb") as file_obj:
```

Comment:
Not directly related to this PR - does this work if `output_path` is specified but in a directory that doesn't exist yet?

Reply:
It does not - the parent directory is expected to exists or `FileNotFoundError` will be raised